topic_modeling/train_models/train_models.py

```diff
@@ -0,0 +1,141 @@
+import os
+import glob
+import re
+import nltk
+import string
+import pickle
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.decomposition import LatentDirichletAllocation
+from sklearn.feature_extraction import text
+from nltk.tokenize import word_tokenize
+from nltk.stem.porter import PorterStemmer
+from nltk.corpus import stopwords
+from nltk.stem.wordnet import WordNetLemmatizer
+
+nltk.download("stopwords")
+nltk.download("wordnet")
+
+stop = set(stopwords.words('english'))
+exclude = set(string.punctuation)
+lemma = WordNetLemmatizer()
+
+# General TODO
+# 1. Turn this into a function with parameters to
+# enter on the command line
+# 2. Use a sample text file for testing
+
+
+def print_top_words(model, feature_names, n_top_words, num_topics, output_dir):
+    myfile = open(os.path.join(
+            output_dir, 'LDA_dartmouth_topic{0}'.format(num_topics)), 'w')
+    for topic_idx, topic in enumerate(model.components_):
+            message = "Topic #{}: ".format(topic_idx)
+            message += " ".join(
+                [feature_names[i]
+                    for i in topic.argsort()[:-n_top_words - 1:-1]])
+            myfile.write("{}\n".format(message))
+    myfile.close()
+
+
+# need two different ways to set directory to pass
+# Travis CI (because .travis file is called from
+# different directory)
+current_file = os.getcwd().split('/')[-1]
+
+if current_file == 'train_models':
+    base_dir = os.getcwd()
+    data_dir = os.path.join(
+        os.path.normpath(os.getcwd() + os.sep + os.pardir),
+        'scrape_training_data')
+else:
+    base_dir = os.path.join(
+            os.getcwd(), 'topic_modeling', 'train_models')
+    data_dir = os.path.join(
+        os.getcwd(), 'topic_modeling', 'scrape_training_data')
+
+# Put all text from articles into one file,
+# if not already
+if not os.path.exists("all_articles.txt"):
+
+    flist = glob.glob(os.path.join(data_dir, 'the_dartmouth', '*.txt'))
+
+    all_articles = []
+
+    for file in flist:
+
+        if os.stat(file).st_size != 0:
+
+            file_text = open(file)
+            text_1 = file_text.read()
+
+            all_articles.append(text_1)
+
+            file_text.close()
+
+    file = open('all_articles.txt', 'w')
+    for article in all_articles:
+        file.write("{}\n".format(article))
+    file.close()
+
+
+all_articles_original = [line.rstrip('\n') for line in open(
+                        'all_articles.txt')]
+
+porter = PorterStemmer()
+
+all_articles_preprocessed = []
+
+for doc in all_articles_original:
+    s = doc.lower()
+    s = re.sub(r'([^\s\w]|_)+', ' ', s)
+    s = re.sub(r'\s+', ' ', s)  # gets rid of tabs
+    s = re.sub(r'[^A-Za-z0-9]+', ' ', s)
+    s = re.sub(r"\[([A-Za-z0-9_]+)\]", r'', s)  # get rid of words in brackets
```

Comment:
all these manipulations are a **perfect** candidate for a function + unittest which would validate that manipulations are doing the correct job

Reply:
yes, yes, yes. they're now in a function with a test!